src/simplification/SimplifyPatternProperties.ts

```diff
@@ -0,0 +1,80 @@
+import { CommonModel } from '../models';
+import { Schema } from '../models/Schema';
+import { Simplifier } from './Simplifier';
+type Output = Record<string, CommonModel> | undefined;
+
+/**
+ * Find out which common models we should extend
+ * 
+ * @param schema to find extends of
+ */
+export default function simplifyPatternProperties(schema: Schema | boolean, simplifier : Simplifier, seenSchemas: Map<any, Output> = new Map()) : Output {
+  if (typeof schema !== 'boolean') {
+    if (seenSchemas.has(schema)) return seenSchemas.get(schema);
+    const output: Output = {};
+    seenSchemas.set(schema, output);
+  
+    for (const [pattern, patternSchema] of Object.entries(schema.patternProperties || {})) {
+      const newModels = simplifier.simplify(patternSchema);
+      if (newModels.length > 0) {
+        addToPatterns(pattern, newModels[0], schema, output);
+      }
+    }
+
+    //If we encounter combination schemas ensure we recursively find the properties
+    if (simplifier.options.allowInheritance !== true) {
+      //Only merge allOf schemas if we don't allow inheritance
+      combineSchemas(schema.allOf, output, simplifier, seenSchemas, schema);
+    }
+    combineSchemas(schema.oneOf, output, simplifier, seenSchemas, schema);
+    combineSchemas(schema.anyOf, output, simplifier, seenSchemas, schema);
+
+    //If we encounter combination schemas ensure we recursively find the properties
+    combineSchemas(schema.then, output, simplifier, seenSchemas, schema);
+    combineSchemas(schema.else, output, simplifier, seenSchemas, schema);
+
+    return !Object.keys(output).length ? undefined : output;
```

Comment:
```suggestion
    return Object.keys(output).length ? output : undefined;
```

Reply:
Gonna change it for https://github.com/asyncapi/generator-model-sdk/blob/2f1c8f57d843d7f97452276acfba26e879f3def7/src/simplification/SimplifyProperties.ts#L39 as well then.